helm/external-dns-app/templates/_helpers.tpl

```diff
@@ -22,5 +22,48 @@ helm.sh/chart: {{ include "chart" . | quote }}
 Selector labels
 */}}
 {{- define "labels.selector" -}}
-app: {{ .Values.name | quote }}
+app: {{ .Release.Name | quote }}
 {{- end -}}
+
+{{/*
+Create the list of domains to update
+*/}}
+{{- define "domain.list" -}}
+{{- if .Values.externalDNS.domainFilterList -}}
+{{- range .Values.externalDNS.domainFilterList }}
+{{ printf "- --domain-filter=%s" . }}
+{{- end }}
+{{- else }}
+{{- if eq .Values.aws.access "external" }}
+{{- printf "- --domain-filter=%s" .Values.aws.baseDomain }}
+{{- else }}
+{{- printf "- --domain-filter=%s" .Values.baseDomain }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Set the zone type when running on AWS
+*/}}
+{{- define "zone.type" -}}
+{{- if eq .Values.aws.access "external" }}
+{{ printf "- --aws-zone-type=%s" "public" }}
+{{- else }}
+{{- if .Values.aws.zoneType }}
+{{ printf "- --aws-zone-type=%s" .Values.aws.zoneType }}
+{{- end }}
+{{- end }}
+{{- end }}
```

Comment:
if the aws access type is external (utilising credentials to talk to aws instead of using kiam) then we can only update public zones, so we enforce that. if access is internal and the zoneType is provided by the user then we use that, otherwise we don't add the flag at all.

Reply:
I feel this is a bit confusing, why is it that if we want to provide our own AWS access token that we can only modify public zones?

Reply:
it is just a filter as far as I know. So you can enforce in IAM too. The point it when you set credentials (`external`) then you cannot modify private zones

Reply:
@jw-s Fer is correct - if you're using external auth then you are only able to update public zones (which is why this default is enforced). Only by using KIAM can you update private zones too.